### ISSUE TITLE
missing `-` in values tpl helpers can cause incorrect generation of values files path

### ISSUE THREAD

**Issue by chrisdoherty4** [helmfile/helmfile#1673]
### Operating system

Darwin

### Helmfile Version

v0.167.1

### Helm Version

v3.15.3

### Bug description

When defining template helpers for helmfile YAMLs as noted in https://helmfile.readthedocs.io/en/latest/templating_funcs/#include, if the author doesn't include a `-` in the trailing `}}` it can cause the values file name to be created with whitespace causing a `helm` invocation error.

A minimal example of the bug can be found at https://github.com/chrisdoherty4/exp/tree/main/helmfile-tpl-bug.

[It requires a `values` to be specified in the release and a `_*.tpl` to be specified with the missing `-`](https://github.com/chrisdoherty4/exp/blob/main/helmfile-tpl-bug/_helper.tpl#L3).

[I tracked down the code that renders a file path](https://github.com/helmfile/helmfile/blob/6374c6368d01f307e81b5cec6c0ae47b44897db0/pkg/state/temp.go#L61) and noticed it references the namespace. When I specify a namespace the path renders properly so it seems like the namespace is getting some whitespace.

### Example helmfile.yaml

```
releases:
- name: foo
  chart: foo
  version: 0.0.0
  values:
  - values.yml
```

### Error message you've seen (if any)

```
helmfile-tpl-bug > helmfile template -f ./helmfile.yml
Building dependency release=foo, chart=foo
Templating release=foo, chart=foo
in ./helmfile.yml: command "/opt/homebrew/bin/helm" exited with non-zero status:

PATH:
  /opt/homebrew/bin/helm

ARGS:
  0: helm (4 bytes)
  1: template (8 bytes)
  2: foo (3 bytes)
  3: foo (3 bytes)
  4: --version (9 bytes)
  5: 0.0.0 (5 bytes)
  6: --namespace (11 bytes)
  7:    (1 bytes)
  8: --values (8 bytes)
  9: /var/folders/7v/cwtc1d395m505pcqykx5vh_c0000gn/T/helmfile3974500205/  -foo-values-7575495bd9 (91 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/chris_doherty/.kube/config
  WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/chris_doherty/.kube/config
  Error: read /var/folders/7v/cwtc1d395m505pcqykx5vh_c0000gn/T/helmfile3974500205/: is a directory

COMBINED OUTPUT:
  WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/chris_doherty/.kube/config
  WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/chris_doherty/.kube/config
  Error: read /var/folders/7v/cwtc1d395m505pcqykx5vh_c0000gn/T/helmfile3974500205/: is a directory
```

### Steps to reproduce

https://github.com/chrisdoherty4/exp/tree/main/helmfile-tpl-bug

### Working Helmfile Version

N/A

**Comment by chrisdoherty4** [comment]
I tracked this down. 

1. We [`RenderTemplateContentToString`](https://github.com/helmfile/helmfile/blob/6374c6368d01f307e81b5cec6c0ae47b44897db0/pkg/state/release.go#L38) for the ReleaseSpec.Namespace.
2. [`newTemplate` gets called](https://github.com/helmfile/helmfile/blob/6374c6368d01f307e81b5cec6c0ae47b44897db0/pkg/tmpl/context_tmpl.go#L129) in `RenderTemplateToBuffer`
3. [The `_*.tpl` files are loaded](https://github.com/helmfile/helmfile/blob/6374c6368d01f307e81b5cec6c0ae47b44897db0/pkg/tmpl/context_tmpl.go#L109) so they're available when specifying the `namespace` in YAML.

Because the `_*.tpl` files are loaded and there's whitespace at the end of my particular YAML it results in whitespace being included for the namespace even though its unset. That snowballs because all the `if release.Namespace != ""` checks are now invalid.

I think the fix is, for a subset of the `ReleaseSpec` fields, we should trim whitespace after rendering. For example both `Name` and `Namespace` could probably do with whitespace trimming.

**Comment by chrisdoherty4** [comment]
Potential fix in #1674

**Comment by chrisdoherty4** [comment]
Another option would be to only render the `Namespace` (and other fields) of `release.Namespace != ""`.
